crates/ark/src/data_explorer/column_profile.rs

```diff
@@ -0,0 +1,289 @@
+//
+// column_profile.rs
+//
+// Copyright (C) 2024 by Posit Software, PBC
+//
+//
+
+use amalthea::comm::comm_channel::CommMsg;
+use amalthea::comm::data_explorer_comm::ColumnFrequencyTable;
+use amalthea::comm::data_explorer_comm::ColumnHistogram;
+use amalthea::comm::data_explorer_comm::ColumnProfileParams;
+use amalthea::comm::data_explorer_comm::ColumnProfileRequest;
+use amalthea::comm::data_explorer_comm::ColumnProfileResult;
+use amalthea::comm::data_explorer_comm::ColumnProfileSpec;
+use amalthea::comm::data_explorer_comm::ColumnProfileType;
+use amalthea::comm::data_explorer_comm::ColumnSummaryStats;
+use amalthea::comm::data_explorer_comm::DataExplorerFrontendEvent;
+use amalthea::comm::data_explorer_comm::FormatOptions;
+use amalthea::comm::data_explorer_comm::GetColumnProfilesParams;
+use amalthea::comm::data_explorer_comm::ReturnColumnProfilesParams;
+use amalthea::socket::comm::CommSocket;
+use anyhow::anyhow;
+use harp::exec::RFunction;
+use harp::exec::RFunctionExt;
+use harp::tbl_get_column;
+use harp::RObject;
+use harp::TableKind;
+use stdext::unwrap;
+
+use crate::data_explorer::histogram;
+use crate::data_explorer::summary_stats::summary_stats;
+use crate::data_explorer::table::Table;
+use crate::data_explorer::utils::display_type;
+use crate::modules::ARK_ENVS;
+
+pub struct ProcessColumnsProfilesParams {
+    pub table: Table,
+    pub indices: Option<Vec<i32>>,
+    pub kind: TableKind,
+    pub request: GetColumnProfilesParams,
+}
+
+pub async fn handle_columns_profiles_requests(
+    params: ProcessColumnsProfilesParams,
+    comm: CommSocket,
+) -> anyhow::Result<()> {
+    let event = process_columns_profiles_requests(params).await?;
+    let json_event = serde_json::to_value(event)?;
+    comm.outgoing_tx.send(CommMsg::Data(json_event))?;
```

Comment:
I'm still confused that we don't reply in case of errors, since there is a callback ID I would expect the frontend is keeping track of it. But I guess this is cleaned up when it no longer needs the answer because the data has changed? (I haven't looked into the frontend code yet)

Reply:
Yeah, currently the RPC contract does not implement any error handling, so if it fails and we never respond, all the front-end will see is a timeout. 

I added some handling here, so basically, when it fails we will return an empty result set:

https://github.com/posit-dev/ark/blob/56726b7ae1ceda0224efdda20435c2bc70adac25/crates/ark/src/data_explorer/column_profile.rs#L57-L66

We already return empty profiles if we fail to compute profiles, thus it seems reasonable to also do that if everything fails. 

We can add error handling too, we would need make `ReturnColumnsProfile` an enum so we would return either the actual profiles or an error message, but I'd rather add this in another PR because it will also require changes in the python backend.

Reply:
I created https://github.com/posit-dev/positron/issues/4559 to track this issue.